uniflow/model/model.py

```diff
@@ -33,40 +33,39 @@ def __init__(
         """
         model_server_cls = ModelServerFactory.get(model_config["model_server"])
         self._model_server = model_server_cls(model_config)
-        if isinstance(guided_prompt_template, GuidedPrompt):
-            self._guided_prompt_template = guided_prompt_template.get_prompt()
-        else:
-            self._guided_prompt_template = guided_prompt_template
+        self._guided_prompt_template = guided_prompt_template
         self._prompt_keys = []
 
     def _serialize(self, data: List[Context]) -> List[str]:
         """Serialize data.
 
         Args:
-            data (List[Dict[str, Any]]): Data to serialize.
+            data (List[Context]): Data to serialize.
 
         Returns:
             List[str]: Serialized data.
         """
         output = []
         for d in data:
+            if not isinstance(d, Context):
+                if "context" in d:
+                    d = Context(context=d["context"])
+                else:
+                    raise ValueError(
+                        "Input data must be a Context object or have a 'context' field."
+                    )
```

Comment:
This is a very bad practice, you should try to directly cast d into Context by doing d = Context(**d)... 

Also, another problem is that are you trying to handle both data List[Context | Dict[str, Any]] cases?